--- a/go-jsonnet/cmd.go
+++ b/go-jsonnet/cmd.go
@@ -14,9 +14,11 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package main
+package gojsonnet
 
 import (
+	"github.com/git-justanotherone/jsonnet-custodian/cmd/internal/upstream/go-jsonnet/cmd"
+	"github.com/git-justanotherone/jsonnet-custodian/cmd/internal/utils"
 	"fmt"
 	"io"
 	"os"
@@ -27,7 +29,6 @@ import (
 	"github.com/fatih/color"
 
 	"github.com/google/go-jsonnet"
-	"github.com/google/go-jsonnet/cmd/internal/cmd"
 )
 
 func version(o io.Writer) {
@@ -42,8 +43,6 @@ func usage(o io.Writer) {
 	fmt.Fprintln(o, "Available options:")
 	fmt.Fprintln(o, "  -h / --help                This message")
 	fmt.Fprintln(o, "  -e / --exec                Treat filename as code")
-	fmt.Fprintln(o, "  -J / --jpath <dir>         Specify an additional library search dir")
-	fmt.Fprintln(o, "                             (right-most wins)")
 	fmt.Fprintln(o, "  -o / --output-file <file>  Write to the output file rather than stdout")
 	fmt.Fprintln(o, "  -m / --multi <dir>         Write multiple files to the directory, list files")
 	fmt.Fprintln(o, "                             on stdout")
@@ -75,14 +74,6 @@ func usage(o io.Writer) {
 	fmt.Fprintln(o, "                                    var <var>")
 	fmt.Fprintln(o, "  --tla-code-file <var>=<file>      Read the code from the file")
 	fmt.Fprintln(o)
-	fmt.Fprintln(o, "Environment variables:")
-	fmt.Fprintln(o, "  JSONNET_PATH is a colon (semicolon on Windows) separated list of directories")
-	fmt.Fprintln(o, "  added in reverse order before the paths specified by --jpath (i.e. left-most")
-	fmt.Fprintln(o, "  wins). E.g. these are equivalent:")
-	fmt.Fprintln(o, "    JSONNET_PATH=a:b jsonnet -J c -J d")
-	fmt.Fprintln(o, "    JSONNET_PATH=d:c:a:b jsonnet")
-	fmt.Fprintln(o, "    jsonnet -J b -J a -J c -J d")
-	fmt.Fprintln(o)
 	fmt.Fprintln(o, "In all cases:")
 	fmt.Fprintln(o, "  <filename> can be - (stdin)")
 	fmt.Fprintln(o, "  Multichar options are expanded e.g. -abc becomes -a -b -c.")
@@ -95,7 +86,6 @@ type config struct {
 	outputFile           string
 	evalMultiOutputDir   string
 	inputFiles           []string
-	evalJpath            []string
 	filenameIsCode       bool
 	evalMulti            bool
 	evalStream           bool
@@ -107,7 +97,6 @@ func makeConfig() config {
 		filenameIsCode: false,
 		evalMulti:      false,
 		evalStream:     false,
-		evalJpath:      []string{},
 	}
 }
 
@@ -196,12 +185,6 @@ func processArgs(givenArgs []string, config *config, vm *jsonnet.VM) (processArg
 				return processArgsStatusFailure, fmt.Errorf("invalid --max-stack value: %d", l)
 			}
 			vm.MaxStack = l
-		} else if arg == "-J" || arg == "--jpath" {
-			dir := cmd.NextArg(&i, args)
-			if len(dir) == 0 {
-				return processArgsStatusFailure, fmt.Errorf("-J argument was empty string")
-			}
-			config.evalJpath = append(config.evalJpath, dir)
 		} else if arg == "-V" || arg == "--ext-str" {
 			if err := handleVarVal(vm.ExtVar); err != nil {
 				return processArgsStatusFailure, err
@@ -391,7 +374,7 @@ func writeOutputStream(output []string, outputFile string) (err error) {
 	return nil
 }
 
-func main() {
+func CmdJsonnetMain(args []string) error {
 	cmd.StartCPUProfile()
 	defer cmd.StopCPUProfile()
 
@@ -399,12 +382,8 @@ func main() {
 	vm.ErrorFormatter.SetColorFormatter(color.New(color.FgRed).Fprintf)
 
 	config := makeConfig()
-	jsonnetPath := filepath.SplitList(os.Getenv("JSONNET_PATH"))
-	for i := len(jsonnetPath) - 1; i >= 0; i-- {
-		config.evalJpath = append(config.evalJpath, jsonnetPath[i])
-	}
 
-	status, err := processArgs(os.Args[1:], &config, vm)
+	status, err := processArgs(args, &config, vm)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, "ERROR: "+err.Error())
 	}
@@ -426,9 +405,8 @@ func main() {
 		os.Exit(1)
 	}
 
-	vm.Importer(&jsonnet.FileImporter{
-		JPaths: config.evalJpath,
-	})
+	// Configure VM with extensions
+	utils.ConfigureVMExtensions(vm)
 
 	if len(config.inputFiles) != 1 {
 		// Should already have been caught by processArgs.
@@ -485,5 +463,5 @@ func main() {
 			os.Exit(1)
 		}
 	}
-
+	return nil
 }
